src/main/kotlin/calculator/Operator.kt

```diff
@@ -0,0 +1,15 @@
+package calculator
+
+enum class Operator(val symbol: String, val operation: (Double, Double) -> Double) {
+    ADD("+", { a, b -> a + b }),
+    SUBTRACT("-", { a, b -> a - b }),
+    MULTIPLY("*", { a, b -> a * b }),
+    DIVIDE("/", { a, b ->
+        require(b != 0.0) { "Division by zero is not allowed" }
+        a / b
+    }), ;
+
+    companion object {
+        fun fromSymbol(symbol: String): Operator? = entries.find { it.symbol == symbol }
+    }
+}
```

Comment:
It’s really good! I mean, you should move on to the next step and try to reflect my feedback in your code